features/nanostack/FEATURE_NANOSTACK/nanostack-interface/NanostackInterface.cpp

```diff
@@ -454,13 +439,20 @@ NanostackInterface * NanostackInterface::get_stack()
     return _ns_interface;
 }
 
-
 const char * NanostackInterface::get_ip_address()
 {
+    static char text_address[40];
```

Comment:
test_address should probably be a part of the NanostackInterface class rather than static. You should also probably be holding nanostack's lock during this call.